src/qutip_tensornetwork/core/data/network.py

```diff
@@ -0,0 +1,642 @@
+# Copyright 2019 The TensorNetwork Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This file has been derived from the quantum/quantum.py file in the
+google/TensorNetwork repository. The above license corresponds to the one found
+in the original file. The original file has been extensively modified and these
+modifications follow the same license as the rest of the project does."""
```

Comment:
```suggestion
#
# Note:
#
# This file has been derived from quantum.py in the google/TensorNetwork GitHub repository,
# https://github.com/google/TensorNetwork/blob/9b48cdd98777e4d7f26cccf29a562afe61e51da4/
# tensornetwork/quantum/quantum.py.
#
# The license above is the one from the original file. The file has been extensively modified. In
# addition to the license above, the modifications are also provided under the license in
# LICENSE.txt.
```

Reply:
I suggested changing this to a comment and I modified the text a bit to add an explicit link to the source file and to explicitly mention where the other license is.

Otherwise looks good!